test/unit/model/scope.test.ts

```diff
@@ -0,0 +1,198 @@
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import Knex from 'knex'
+import Config from '../../../config/knexfile'
+import ScopeDB, { Scope } from '../../../src/model/scope'
+import { Consent } from '../../../src/model/consent'
+
+/*
+ * Mock Consent Resource
+ */
+const partialConsent: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123'
+}
+
+/*
+ * Mock Scope Resources
+ */
+const tempScope: Scope = {
+  consentId: '1234',
+  action: 'transfer',
+  accountId: 'sjdn-3333-2123'
+}
+
+const tempScope2: Scope = {
+  consentId: '1234',
+  action: 'balance',
+  accountId: 'sjdn-q333-2123'
+}
+
+/*
+ * Scope Resource Model Unit Tests
+ */
+describe('src/model/scope', (): void => {
+  let Db: Knex
+  let scopeDB: ScopeDB
+
+  beforeAll(async (): Promise<void> => {
+    Db = Knex(Config.test)
+    await Db.migrate.latest()
+    // Enable Sqlite foreign key support
+    await Db.raw('PRAGMA foreign_keys = ON')
+
+    scopeDB = new ScopeDB(Db)
+  })
+
+  afterAll(async (): Promise<void> => {
+    Db.destroy()
+  })
+
+  // Reset table for new test
+  beforeEach(async (): Promise<void> => {
+    await Db<Consent>('Consent').del()
+    await Db<Scope>('Scope').del()
+  })
+
+  describe('register', (): void => {
+    it('adds scope for an existing consent to the database', async (): Promise<void> => {
+      // Setup
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+
+      await scopeDB.register(tempScope)
+
+      // Assertion
+      const scopes: Scope[] = await Db<Scope>('Scope')
+        .select('*')
+        .where({
+          consentId: partialConsent.id
+        })
+
+      expect(scopes.length).toEqual(1)
+      expect(scopes[0].id).toEqual(expect.any(Number))
+      expect(scopes[0]).toEqual(expect.objectContaining(tempScope))
+    })
+
+    it('adds multiple scopes for an existing consent to the database', async (): Promise<void> => {
+      // Setup
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+
+      await scopeDB.register([tempScope, tempScope2])
+
+      // Assertion
+      const scopes: Scope[] = await Db<Scope>('Scope')
+        .select('*')
+        .where({
+          consentId: partialConsent.id
+        })
+
+      expect(scopes.length).toEqual(2)
+      expect(scopes[0].consentId === scopes[1].consentId).toEqual(true)
+      expect(scopes[0].action !== scopes[1].action).toEqual(true)
+    })
+
+    it('returns an error on adding a scope for a non-existent consent', async (): Promise<void> => {
+      await expect(scopeDB.register(tempScope)).rejects.toThrowError()
```

Comment:
Check for the specific error class here?

Reply:
This is a constraint violation. Hence, the error is DB dependent and not defined by the code.
We don't have an error class for this.